base/applications/rapps/appview.cpp

```diff
@@ -1525,6 +1516,30 @@ CApplicationView::ProcessWindowMessage(
             OnCommand(wParam, lParam);
         }
         break;
+
+        case WM_CONTEXTMENU:
+        {
+            bool kbd = -1 == (int)(INT_PTR)lParam;
+            if ((HWND)wParam == m_ListView->m_hWnd)
+            {
+                int item = ListView_GetNextItem((HWND)wParam, -1, LVNI_FOCUSED|LVNI_SELECTED);
+                if (item != -1)
+                {
+                    POINT *ppt = NULL, pt;
+                    if (kbd)
+                    {
+                        RECT r;
+                        ListView_GetItemRect((HWND)wParam, item, &r, LVIR_LABEL);
+                        pt.x = r.left + (r.right - r.left) / 2;
+                        pt.y = r.top + (r.bottom - r.top) / 2;
+                        ::ClientToScreen((HWND)wParam, ppt = &pt);
+                    }
+                    ShowPopupMenuEx(m_hWnd, m_hWnd, 0, ID_INSTALL, ppt);
+                    return TRUE;
+                }
+            }
+        }
+        break;
```

Comment:
It looks quite more complicated than it was previously, is it really worth it?

Reply:
I'd argue that yes, you are supposed to be able to access the context menu with the keyboard even if the commands are available on the toolbar.

It is not that complicated, 3 extra function calls.